compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ViewControllerBasedLifecycleOwner.uikit.kt

```diff
@@ -19,9 +19,11 @@ package androidx.compose.ui.window
 import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.LifecycleOwner
 import androidx.lifecycle.LifecycleRegistry
-import kotlin.test.fail
+import platform.Foundation.NSNotificationCenter
 
-internal class ViewControllerBasedLifecycleOwner : LifecycleOwner {
```

Comment:
The main suggestion in https://github.com/JetBrains/compose-multiplatform-core/pull/1198 is to get rid of the whole state machine. It is not needed as the lifecycle itself takes care of it. The result code will be much simpler (I post a snippet there)

Reply:
I've removed the state machine and removed all checks as they don't really belong there. It allowed to simplify code even further.

Reply:
We have to remember intermediate state as in the snippet. Because we can't guarantee the order of `ApplicationStateListener` and `handleViewWillAppear`. If `ApplicationStateListener` happens before `handleViewWillAppear`, we will have an invalid state.

Reply:
The same should be done for other platforms, we have this issue on them as well.

Reply:
@igordmn I've changed the handling of events, please re-review